tests/compile-fail/if-not-linear.rs

```diff
@@ -0,0 +1,37 @@
+#![feature(plugin)]
+#![feature(custom_attribute)]
+#![plugin(humpty_dumpty)]
+#![allow(unused_attributes)]
+#![allow(unused_variables)]
+#![allow(dead_code)]
+
+#[drop_protect]
+struct Foo;
+
+impl Foo {
+    #[allow_drop="Foo"]
+    fn close(self) { }
+}
+
+fn test1() {
+    let f = Foo; //~ ERROR dropped var
+    if false { return } else { f.close(); }
+    //~^ ERROR Branch arms are not linear
+}
+
+fn test2() {
+    let f = Foo;
```

Comment:
Note how the inversion of the branches from `test1` to `test2` drops the `dropped var` error